sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -31,9 +31,11 @@ class DDLSuite extends QueryTest with SharedSQLContext {
    */
   private def withDatabase(dbNames: String*)(f: => Unit): Unit = {
     try f finally {
+      val currentDb = sqlContext.sessionState.catalog.getCurrentDatabase
       dbNames.foreach { name =>
         sqlContext.sql(s"DROP DATABASE IF EXISTS $name CASCADE")
       }
+      sqlContext.sessionState.catalog.setCurrentDatabase(currentDb)
```

Comment:
@gatorsmile Yeah.,initially i was thinking of resetting to "default". I thought a restore semantics might be better here. Is it possible for us to do nesting in withDatabase ? Like
WithDatabase("foo") {
   // create db
   // sql
   // withDatabase("bar") {
        sql
      }
      sql
      sql
}
Let me know if you want to reset it to "default" instead of restoring ?